syntax = "proto2";

package PolicyProto;

message ConfigurationSettings
{
    repeated string sophosCdnURLs = 1;
    repeated string updateCache = 2;
    repeated string messageRelay = 25;

    message ProxyCredentialT
    {
        optional string username = 1;
        optional string password = 2;
        optional string proxyType = 3;
    }
    message ProxyT
    {
        optional string url = 1;
        optional ProxyCredentialT credential = 2;
    }


    optional ProxyT proxy = 4;
    optional string JWToken = 5;
    optional string tenantId = 6;
    optional string deviceId = 7;
    optional string versigPath = 8;
    repeated string installArguments = 9;

    enum LogLevelOption
    {
       NORMAL = 0;
       VERBOSE = 1;
    }

    optional LogLevelOption logLevel = 10 [default=NORMAL];
    optional bool forceReinstallAllProducts = 11 [default=false];
    repeated string manifestNames = 12;
    repeated string optionalManifestNames = 13;

    message Subscription
    {
        optional string rigidName = 1;
        optional string baseVersion = 2;
        optional string tag = 3;
        optional string fixedVersion = 4;
    }

    optional Subscription primarySubscription = 14;
    repeated Subscription products = 15;
    repeated string features = 16;

    /**
        Configures SulDownloader to download from slow supplements, where such a choice is available.
     */
    optional bool useSlowSupplements = 17 [default=false];


    optional string updateCacheCertPath = 19;
    optional bool forceUpdate = 20 [default=false];
    optional bool forcePausedUpdate = 21 [default=false];

    message ESMVersion
    {
        optional string token = 1;
        optional string name = 2;
    }
    optional ESMVersion esmVersion = 22;

    optional string sophosSusURL = 23;
    optional bool useSdds3DeltaV2 = 24 [default=false];
    optional bool useSdds3DeltaV3 = 26 [default=false];
}
